### PR TITLE
Fix sonar workflow

### PR DESCRIPTION
**Pull request by EvaristeGalois11**
Ok this is it, I can feel it :D

I made the following improvements from the previous attempt: 

- Uploading every target directory instead of just the test coverage. It's a zstandard compressed archive to save storage space, it cuts down almost 90% of the size this way.
- Installing every module in the local maven repository. I personally tested that even with the .class files sonar still complains about the lack of modules installed inside the .m2 folder. I don't know if it's a deal breaker for the analysis but to be extra safe about it i installed every module to shut it up. It should be safe to do so because no maven life cycle is started, it's just an explicit goal invocation of a very precise plugin. It shouldn't be possible to sneak some malicious plugin this way.
- Deleting the artifact after downloading it. This should avoid an unnecessary clutter of the storage space.
- This time i actually tested it in my forked environment lol https://github.com/EvaristeGalois11/instancio/actions It seems to reach the SONAR_TOKEN problem without any other weird errors

### CODE REVIEW

**Comment by armandino** [comment]
@EvaristeGalois11 although this may work, I'm not sure we should go down this route. It would require modifying the workflow whenever a test module is added, removed, or renamed.

File: .github/workflows/sonar.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+          mvn -B install:install-file -Dfile=instancio-tests/java16-tests/target/java16-tests-"${VERSION}".jar -DpomFile=instancio-tests/java16-tests/pom.xml
+          mvn -B install:install-file -Dfile=instancio-tests/java17-tests/target/java17-tests-"${VERSION}".jar -DpomFile=instancio-tests/java17-tests/pom.xml
+          mvn -B install:install-file -Dfile=instancio-tests/kotlin-tests/target/kotlin-tests-"${VERSION}".jar -DpomFile=instancio-tests/kotlin-tests/pom.xml
+          mvn -B install:install-file -Dfile=instancio-tests/packaging-tests/target/packaging-tests-"${VERSION}".jar -DpomFile=instancio-tests/packaging-tests/pom.xml
+          mvn -B install:install-file -Dfile=instancio-tests/report-aggregate/pom.xml -DgroupId=org.instancio -DartifactId=report-aggregate -Dversion="${VERSION}" -Dpackaging=pom
+          mvn -B install:install-file -Dfile=instancio-tests/spi-old-tests/target/spi-old-tests-"${VERSION}".jar -DpomFile=instancio-tests/spi-old-tests/pom.xml
+          mvn -B install:install-file -Dfile=instancio-tests/spi-tests/target/spi-tests-"${VERSION}".jar -DpomFile=instancio-tests/spi-tests/pom.xml
```
Comment (armandino) [inline .github/workflows/sonar.yml:86]: Let's see if we can do this without explicitly listing all the modules.

**Comment by EvaristeGalois11** [comment]
I understand your concern @armandino

I don't think we can tell maven to install all the jars without explicitly listing them, but what if we just zip the previous local repository? The path `~/.m2/org/instancio/` should already contain all the artifacts installed anyway, we can simply move them from the Build workflow to the Sonar workflow.

It shouldn't be necessary to list all the modules anymore. Could it work for you?

**Comment by armandino** [comment]
@EvaristeGalois11 worth a try. I don't know what the best approach is, but it sounds better than listing the modules explicitly. Are there are any downsides to doing this?

**Comment by EvaristeGalois11** [comment]
Nothing i can think of, it's probably way safer and cleaner to do it this way instead of installing them manually.

I'm testing it in my repository, i'll update the PR if everything works.

**Comment by EvaristeGalois11** [comment]
Ok this is in fact much more  simple lol

Please @armandino take a look

The only assumption that I made to create the repository-artifact is the `org.instancio` group name, but I think it's much more manageable than the list of all the modules. What do you think?

**Comment by armandino** [comment]
@EvaristeGalois11 looks good to me! Let's give it a try.

**Pull request merged**